src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+                System.out.println("Okay, I've deleted that task!");
+                System.out.println(tasks.get(taskIndex - 1));
+                tasks.remove(taskIndex - 1);
+            } catch (NumberFormatException e) {
+                throw new IllegalCommandException();
+            }
+        }
+        saveData();
+    }
+
+    public static void startDuke() throws IOException {
+        if (Files.notExists(Paths.get(FILE_PATH))) {
+            Files.createDirectories(Paths.get(FILE_PATH).getParent());
+            Files.createFile(Paths.get(FILE_PATH));
+            System.out.println("Hi! I'm Duke. I've created your data file for you, what would you like me to do?");
+        } else {
+            loadData();
+            System.out.println("Welcome back!");
+            commandList();
+        }
+    }
+
+    public static void loadData() throws FileNotFoundException {
+        File f = new File(FILE_PATH);
+        Scanner s = new Scanner(f);
+        while(s.hasNext()) {
+            tasks.add(parseTask(s.nextLine()));
+        }
+    }
+
+    public static void saveData() throws IOException {
+        FileWriter fw = new FileWriter(FILE_PATH);
+        for (Task task : tasks) {
+            fw.write(task.toText() + "\n");
+        }
+        fw.close();
+    }
+
+    public static Task parseTask(String taskInfo) throws IllegalArgumentException {
+        String[] infoParsed = taskInfo.split("\\|");
+        Task newTask;
+
+        switch (infoParsed[0]) {
+            case "[D] ":
+                newTask = new Deadline(infoParsed[2].strip(), infoParsed[3].strip());
+                break;
+            case "[E] ":
+                newTask = new Event(infoParsed[2].strip(), infoParsed[3].strip());
+                break;
+            case "[T] ":
+                newTask = new Todo(infoParsed[2].strip());
+                break;
+            default:
+                throw new IllegalArgumentException();
+        }
+
+        if (infoParsed[1].equals(" [X] ")) {
+            newTask.setDone(true);
+        }
+        return newTask;
+    }
+
+    public static void main(String[] args) {
+        try {
+            startDuke();
+        } catch (IOException e) {
+            System.out.println("I could not create/load your file!");
+            System.exit(1);
+        }
+        Scanner in = new Scanner(System.in);
+        // Read a command from the user.
+        String line = in.nextLine();
+        // Check which command the user entered and
+        // call the corresponding method.
+        while (!line.equals("bye")) {
+            try {
+                if (line.equals("list")) {
+                    commandList();
+                } else if (line.startsWith("done")) {
+                    doneTask(line);
+                } else if (line.startsWith("delete")) {
+                    deleteTask(line);
+                } else {
+                    addTask(line);
+                }
+            //exception handling
```

Comment:
This is obvious and does not need a comment. We try to minimize comments for obvious code